src/main/java/chess/domain/square/File.java

```diff
@@ -0,0 +1,49 @@
+package chess.domain.square;
+
+import java.util.Arrays;
+
+public enum File {
+
+    a(1),
+    b(2),
+    c(3),
+    d(4),
+    e(5),
+    f(6),
+    g(7),
+    h(8);
+
+    static final String ERROR_NOT_EXIST_FILE = "은(는) 존재하지 않는 파일입니다.";
```

Comment:
```suggestion
    private static final String ERROR_NOT_EXIST_FILE = "은(는) 존재하지 않는 파일입니다.";
```
자바에서는 protected와 default를 제공하지만 완전한 접근제어자가 아닌 결국 접근이 가능한 접근제어자입니다. 미션을 진행하면서 가능하면 public과 private으로만 설계할 수 없는지 고민해보세요.

테스트에서 사용되는 경우 테스트 결과값은 클래스에서 사용하는 상수를 그대로 사용하기보단 무엇을 검증하는지는 테스트에게 위임해보세요.

Reply:
말씀해 주신 대로 가능하면 public과 private 접근 제어자만 사용하기 위해 
일단 기존 예외 메시지를 검증하는 코드는 지우고, 해당 예외 메시지 상수들도 모두 private로 변경해 주었습니다.

그런데 `무엇을 검증할지는 테스트에게 맡긴다` 의 의미가 잘 이해되지 않습니다.

만약, 아래와 같이 검증 메서드가 많고 모두 `IllegalArgumentException`을 발생시킨다고 가정했을 때,
```java
public void move(final Square source, final Square target, final PieceColor turn) {
        validateIsSameSquare(source, target);
        validateIsNonExistentPiece(source);

        Piece sourcePiece = findPieceBySquare(source);
        validateIsTurn(sourcePiece, turn);
        validateCanMove(source, target, sourcePiece);
        validateExistObstacleOnPath(source, target);
        pieces.remove(source);
        pieces.put(target, sourcePiece);
    }
```

예외가 정말로 발생하는지 테스트에게 맡기려면 아래 코드 정도로만 작성해도 충분한가요?
```java
assertThatThrownBy(() -> board.move(source, target, turn))       
    .isInstanceOf(IllegalArgumentException.class);
```

어느 검증 메서드에서 발생한 예외인지 예외 메시지를 검증을 통해 명확히 알고 싶다면 어떻게 하면 될까요??  테스트 메서드 안에서 예외 메시지 값을 하드 코딩 해주거나 상수로 또 정의하는 게 좋을까요?

Reply:
assertion에서 검증하는 부분이 production에 의존적이지 않아야한다는 이야기였어요.

예외의 메시지를 테스트하는 경우 저희는 `ERROR_NOT_EXIST_FILE`가 들어가는 것을 기대하는 것이 아니라 `은(는) 존재하지 않는 파일입니다.`라는 문구가 나오는지 테스트하기를 기대해요. 즉, 테스트는 production의 값을 검증하기 때문에 production 코드로 영향이 가는 테스트를 작성하면 안되는거죠.

TDD를 예를들면 테스트를 먼저 작성했더라면 `ERROR_NOT_EXIST_FILE`를 사용하는 테스트가 나왔었을까요? 제 생각에는 `"은(는) 존재하지 않는 파일입니다."`라는 String값을 테스트에서 사용했을 것 같아요. 테스트 케이스는 운영의 코드에 상수가 들어간다는 사실을 모르고 있으니까요.

https://jojoldu.tistory.com/615 글을 한번 참고해보시면 도움이 될 것 같네요 😄